src/tools/singlejar/port.h

```diff
@@ -19,13 +19,27 @@
 // This macro is required to tell MSVC headers to also define POSIX names
 // without "_" prefix (such as "open" for "_open").
 #define _CRT_DECLARE_NONSTDC_NAMES 1
-#include <fcntl.h>
 #include <io.h>
+#endif
+
+#include <fcntl.h>
 #include <stddef.h>
 #include <stdio.h>
+#include <sys/stat.h>
 #include <sys/types.h>
 #include <time.h>
 
+#if defined(__APPLE__)
+typedef off_t off64_t;
+#elif defined(_WIN32)
+typedef __int64 off64_t;
+#endif
+static_assert(sizeof(off64_t) == 8, "File offset type must be 64-bit");
+
+#ifdef _WIN32
```

Comment:
Wait, this one has no terminating `#endif`

Reply:
https://github.com/rongjiecomputer/bazel/blob/257beec22b632c9535d0c9d6e4218f2dbf3c0593/src/tools/singlejar/port.h

- Line 18 is closed by line 23.
- Line 32 is closed by line 36.
- Line 39 is closed by line 67.
- In the third block, line 43 is closed by line 51.

I need to include `<sys/stat.h>` for all platforms to get `off_t` and `off64_t`.

Reply:
Ah indeed, you added an `#endif` in line 23.